server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -281,18 +281,16 @@ static boolean resolvePipelines(final DocWriteRequest<?> originalRequest, final
             IndexMetaData indexMetaData = metaData.indices().get(originalRequest.index());
             // check the alias for the index request (this is how normal index requests are modeled)
             if (indexMetaData == null && indexRequest.index() != null) {
-                AliasOrIndex indexOrAlias = metaData.getAliasAndIndexLookup().get(indexRequest.index());
-                if (indexOrAlias != null && indexOrAlias.isAlias()) {
-                    AliasOrIndex.Alias alias = (AliasOrIndex.Alias) indexOrAlias;
-                    indexMetaData = alias.getWriteIndex();
+                IndexSpace indexOrAlias = metaData.getAliasAndIndexLookup().get(indexRequest.index());
```

Comment:
We'll probably want to rename that variable from `indexOrAlias`. Can be done in a later PR, though.